src/Psalm/Context.php

```diff
@@ -458,6 +502,28 @@ public function update(
         }
     }
 
+    /**
+     * Updates the list of possible references from a confusing scope,
+     * such as a reference created in an if and then later reused.
+     */
+    public function updateReferencesPossiblyFromConfusingScope(
+        Context $confusing_scope_context,
+        StatementsAnalyzer $statements_analyzer
+    ): void {
+        foreach ($confusing_scope_context->references_in_scope + $confusing_scope_context->references_to_external_scope
+            as $reference_id => $_
+        ) {
```

Comment:
I cannot for the life of me figure out how to make phpcs happy here. I'll probably end up using a temporary variable, but this seems like an issue either in our phpcs configuration or in phpcs itself.

Reply:
```php
foreach ($confusing_scope_context->references_in_scope
                     + $confusing_scope_context->references_to_external_scope as $reference_id => $_
            ) {
            }
```
This makes it happy for me

Reply:
Of course, one of the few possibilities I didn't try. Honestly though, having a multiple line `foreach` without the `as` on its own line just seems more confusing to me. I prefer the extra variable to that.

Reply:
Agree :p